### PR TITLE
Fixed bug in dQuat/dSpq jacobian. Added tests relevant to this issue.

### PR DESCRIPTION
**Pull request by ryanelandt**
The jacobian dQuat_dSPQ is a matrix of zeros for things like `Rotations.jacobian(Quat, SPQuat(1.0, 0.0, 0.0))`.

Note: `SPQuat(1.0, 0.0, 0.0)` is the projection of `Quat(0.0, 1.0, 0.0, 0.0)` which has a non-negative real part.

### CODE REVIEW

**Comment by ryanelandt** [comment]
The changes to `src/derivatives.jl` are the same, but for different reasons. This should be an example of the type of thing that @rdeits meant in [#44 (comment)](https://github.com/FugroRoames/Rotations.jl/issues/44#issuecomment-327860193).

I didn't know if it was okay to do a PR that fails CI. In other words, if I did this PR with just the unit tests would that have been okay?

**Comment by andyferris** [comment]
> if I did this PR with just the unit tests would that have been okay?

That's always a useful contribution, yes. :) You can use `@test_broken` as a way of highlighting a problem (or demand a specific behavior) while not breaking CI.

**Comment by tkoolen** [comment]
Resolved a merge conflict after merging #72.

File: test/derivative_tests.jl [hunk 1]
```diff
@@ -32,7 +32,20 @@ using ForwardDiff
 
                 # test jacobian to a Rotation matrix
                 R_jac = Rotations.jacobian(Quat, spq)
-                FD_jac = ForwardDiff.jacobian(x -> (q = Quat(SPQuat(x[1],x[2],x[3]));
+                FD_jac = ForwardDiff.jacobian(x -> (q = convert(Quat, SPQuat(x[1],x[2],x[3]));
```
Comment (tkoolen) [inline test/derivative_tests.jl]: After #72, `convert(Quat, spquat)` and `Quat(spquat)` should do the same thing. No change needed here, just FYI.

**Comment by tkoolen** [comment]
@andyferris, can I tag a new release?

File: test/derivative_tests.jl (hunk 1 above)
Comment (ryanelandt) [reply test/derivative_tests.jl]: Thanks for the heads up. I'll work on switching over to that once I get used to the new (and improved) behavior.

**Pull request merged**